crates/ide-completion/src/render.rs

```diff
@@ -447,13 +487,54 @@ fn compute_type_match(
         return None;
     }
 
-    if completion_ty == expected_type {
-        Some(CompletionRelevanceTypeMatch::Exact)
-    } else if expected_type.could_unify_with(ctx.db, completion_ty) {
-        Some(CompletionRelevanceTypeMatch::CouldUnify)
-    } else {
-        None
+    match_types(ctx, expected_type, completion_ty)
+}
+
+fn compute_function_type_match(
```

Comment:
As far as I could tell `compute_type_match` does not work for functions as each function has a unique type. Instead  `compute_function_type_match` looks at the params and returns type attempting to unify them. There might already be a way to do this but I couldn't find it.  

` compute_function_type_match` function feels more like a hack/workaround to me but it gets the job done. Perhaps expanding the type unification logic to handle callables could make sense in a separate issue.